### PR TITLE
FIX-#0000: fix cloud examples and Modin in the Cloud

### LINKED ISSUE #3420: Cloud examples are outdated
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: `Ubuntu 20.04`
- **Modin version** (`modin.__version__`): 0a642753af2cd2332ac71166bcddc89b0cace9d5
- **Python version**: `3.8.10`
- **Code we can use to reproduce**:

### Describe the problem

The image in use is no longer available from `AWS` provider, so all examples with the cloud do not work.

UPD: during testing, other types of errors were found, which will be discussed below.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [x] Resolves #3420 
- [ ] tests added and passing
- [ ] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/experimental/cloud/rayscale.py [hunk 1]
```diff
@@ -193,7 +198,7 @@ def _get_modin_version():
         from modin import __version__
 
         # for example: 0.8.0+116.g5e50eef.dirty
-        return f"modin=={__version__.split('+')[0]}"
+        return f"modin-all=={__version__.split('+')[0]}"
```
Comment (anmyachev) [inline modin/experimental/cloud/rayscale.py:201]: To be able to use `pandas_on_ray` and `omnisci_on_native` modes.

File: modin/experimental/cloud/rayscale.py [hunk 2]
```diff
@@ -151,7 +151,12 @@ def __make_config(self):
         res = self._update_conda_requirements(config["setup_commands"][0])
         config["setup_commands"][0] = res
 
-        return bootstrap_config(config)
+        old_setup_commands = config["setup_commands"]
+        old_head_setup_commands = config["head_setup_commands"]
+        bootstraped_config = bootstrap_config(config)
+        bootstraped_config["setup_commands"] = old_setup_commands
+        bootstraped_config["head_setup_commands"] = old_head_setup_commands
+        return bootstraped_config
```
Comment (anmyachev) [inline modin/experimental/cloud/rayscale.py:159]: The problematic side effect is called by `merge_setup_commands` function that leads to the repetition of the settings commands(with multiple `bootstrap_config(config)` calls):
https://github.com/ray-project/ray/blob/77fb273c61dbab2ab2b93101e8649f4eabe7d6ad/python/ray/autoscaler/_private/util.py#L252

File: modin/experimental/cloud/ray-autoscaler.yml [hunk 3]
```diff
@@ -120,9 +120,9 @@ setup_commands:
         source ~/.bashrc
         conda create --name modin --yes
         conda activate modin
-        conda install --yes -c intel/label/modin -c conda-forge {{CONDA_PACKAGES}}
+        conda install --yes -c conda-forge {{CONDA_PACKAGES}}
 
-        pip install colorful cloudpickle
+        pip install cloudpickle rpyc==4.1.5
```
Comment (anmyachev) [inline modin/experimental/cloud/ray-autoscaler.yml:125]: Cloud dependencies do not come in conda packages.

**Comment by anmyachev** [comment]
Faced with new problem: `FileNotFoundError: modin-datasets/trips_data.csv`

File: examples/cluster/taxi-runner.py [hunk 4]
```diff
@@ -51,6 +51,18 @@
...
+
+        OmnisciServer = (
+            get_connection()
+            .modules["modin.experimental.engines.omnisci_on_ray.frame.omnisci_worker"]
+            .OmnisciServer
+        )
+        modin.experimental.engines.omnisci_on_ray.frame.omnisci_worker.OmnisciServer = (
+            OmnisciServer
+        )
+
```
Comment (anmyachev) [inline examples/cluster/taxi-runner.py:65]: Appeared due to Omniscripts changes

**Comment by anmyachev** [comment]
> Faced with new problem: `FileNotFoundError: modin-datasets/trips_data.csv`

Dataset for taxi was be uploaded. Need check.

File: modin/pandas/__init__.py [hunk 5]
```diff
@@ -160,6 +160,7 @@ def init_remote_ray(partition):
         assert (
             Backend.get() == "Omnisci"
         ), f"Backend should be 'Omnisci' with 'Cloudnative' engine, but provided {Backend.get()}."
+        get_connection().modules["modin.config"].IsExperimental.put(True)
```
Comment (anmyachev) [inline modin/pandas/__init__.py:163]: Fix for following error:
```shell
========= Remote Traceback (1) =========
Traceback (most recent call last):
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/data_management/factories/dispatcher.py", line 129, in _update_factory
    cls.__factory = getattr(factories, factory_name)
AttributeError: module 'modin.data_management.factories.factories' has no attribute 'OmnisciOnNativeFactory'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/rpyc/core/protocol.py", line 320, in _dispatch_request
    res = self._HANDLERS[handler](self, *args)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/rpyc/core/protocol.py", line 593, in _handle_call
    return obj(*args, **dict(kwargs))
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/pandas/dataframe.py", line 145, in __init__
    distributed_frame = from_non_pandas(data, index, columns, dtype)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/pandas/utils.py", line 40, in from_non_pandas
    from modin.data_management.factories.dispatcher import FactoryDispatcher
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/data_management/factories/dispatcher.py", line 290, in <module>
    Engine.subscribe(FactoryDispatcher._update_factory)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/config/pubsub.py", line 213, in subscribe
    callback(cls)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/data_management/factories/dispatcher.py", line 150, in _update_factory
    raise FactoryNotFoundError(msg.format(Backend.get(), Engine.get()))
modin.data_management.factories.dispatcher.FactoryNotFoundError: Omnisci on Native is only accessible through the experimental API.
Run `import modin.experimental.pandas as pd` to use Omnisci on Native.

```

File: modin/experimental/cloud/ray-autoscaler.yml [hunk 6]
```diff
@@ -120,9 +122,9 @@ setup_commands:
         source ~/.bashrc
         conda create --name modin --yes
         conda activate modin
-        conda install --yes -c intel/label/modin -c conda-forge {{CONDA_PACKAGES}}
+        conda install --yes -c conda-forge --override-channels {{CONDA_PACKAGES}}
```
Comment (anmyachev) [inline modin/experimental/cloud/ray-autoscaler.yml:125]: Fix for following error:

```shell
========= Remote Traceback (1) =========
Traceback (most recent call last):
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/rpyc/core/protocol.py", line 320, in _dispatch_request
    res = self._HANDLERS[handler](self, *args)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/rpyc/core/protocol.py", line 593, in _handle_call
    return obj(*args, **dict(kwargs))
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/data_management/factories/factories.py", line 599, in prepare
    from modin.experimental.engines.omnisci_on_native.io import OmnisciOnNativeIO
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/experimental/engines/omnisci_on_native/io.py", line 27, in <module>
    from modin.experimental.engines.omnisci_on_native.frame.data import OmnisciOnNativeFrame
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/experimental/engines/omnisci_on_native/frame/data.py", line 18, in <module>
    from .partition_manager import OmnisciOnNativeFramePartitionManager
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/experimental/engines/omnisci_on_native/frame/partition_manager.py", line 23, in <module>
    from .omnisci_worker import OmnisciServer
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/experimental/engines/omnisci_on_native/frame/omnisci_worker.py", line 28, in <module>
    from omniscidbe import PyDbEngine
ImportError: libclang-cpp.so.10: cannot open shared object file: No such file or directory
```

File: modin/experimental/cloud/ray-autoscaler.yml [hunk 7]
```diff
@@ -73,6 +73,7 @@ head_node:
         - DeviceName: /dev/sda1
           Ebs:
               VolumeSize: 40
+    SecurityGroupIds: [sg-0790d73053117d1b0]
```
Comment (anmyachev) [inline modin/experimental/cloud/ray-autoscaler.yml:76]: For the proxy to work correctly during development. Need to delete before merge.

File: modin/experimental/cloud/cluster.py [hunk 8]
```diff
@@ -192,6 +194,7 @@ def __enter__(self):
         self.spawn(wait=True)  # make sure cluster is ready
         self.connection.activate()
         self.old_backends = set_backends(self.target_engine, self.target_partition)
+        Engine.subscribe(_update_engine)
```
Comment (anmyachev) [inline modin/experimental/cloud/cluster.py:197]: Local engine should be updated before working in remote context (if don't do this, unexpected errors appear).

```shell
========= Remote Traceback (1) =========
Traceback (most recent call last):
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/rpyc/core/protocol.py", line 320, in _dispatch_request
    res = self._HANDLERS[handler](self, *args)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/rpyc/core/protocol.py", line 593, in _handle_call
    return obj(*args, **dict(kwargs))
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/pandas/dataframe.py", line 109, in __init__
    Engine.subscribe(_update_engine)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/config/pubsub.py", line 213, in subscribe
    callback(cls)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/pandas/__init__.py", line 113, in _update_engine
    initialize_ray()
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/engines/ray/utils.py", line 119, in initialize_ray
    ray.init(
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/ray/_private/client_mode_hook.py", line 82, in wrapper
    return func(*args, **kwargs)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/ray/worker.py", line 929, in init
    _global_node = ray.node.Node(
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/ray/node.py", line 181, in __init__
    session_name = _get_with_retry(redis_client, "session_name")
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/ray/node.py", line 41, in _get_with_retry
    result = redis_client.get(key)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/client.py", line 1606, in get
    return self.execute_command('GET', name)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/client.py", line 898, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/connection.py", line 1192, in get_connection
    connection.connect()
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/connection.py", line 567, in connect
    self.on_connect()
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/connection.py", line 643, in on_connect
    auth_response = self.read_response()
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/connection.py", line 756, in read_response
    raise response
redis.exceptions.ResponseError: WRONGPASS invalid username-password pair
```

File: examples/cluster/mortgage-runner.py [hunk 9]
```diff
@@ -17,21 +17,25 @@
...
     "aws_credentials",
     cluster_name="rayscale-test",
-    region="eu-north-1",
-    zone="eu-north-1b",
-    image="ami-00e1e82d7d4ca80d3",
+    region="eu-central-1",
+    zone="eu-central-1b",
+    image="ami-05f7491af5eef733a",
 )
+with test_cluster:
```
Comment (vnlitvinov) [inline examples/cluster/mortgage-runner.py:34]: why are you doing this in `with cluster`?

File: modin/experimental/cloud/rayscale.py [hunk 10]
```diff
@@ -151,7 +151,12 @@ def __make_config(self):
         res = self._update_conda_requirements(config["setup_commands"][0])
         config["setup_commands"][0] = res
 
-        return bootstrap_config(config)
+        old_setup_commands = config["setup_commands"]
+        old_head_setup_commands = config["head_setup_commands"]
+        bootstraped_config = bootstrap_config(config)
+        bootstraped_config["setup_commands"] = old_setup_commands
+        bootstraped_config["head_setup_commands"] = old_head_setup_commands
```
Comment (vnlitvinov) [inline modin/experimental/cloud/rayscale.py:158]: this looks weird, please at least add a comment explaining what's going on here

**Comment by anmyachev** [comment]
> Please split the PR at least in two - one fixing issues with "modin in the cloud" itself and another fixing the examples.

Please take a look on https://github.com/modin-project/modin/pull/3564

**Pull request closed without merging**